### ISSUE TITLE
Issues with Winblend and Transparent Terminal Background

### ISSUE THREAD

**Issue by HarthTuwist** [dstein64/nvim-scrollview#112]
I am using Windows Terrminal with nvim and nvim-scrollview, and the kanagawa (dragon) color scheme, but I am having issues with a transparent background in nvim (such that my background image is visible): If winblend is set to something other than 100, it seems to blend the color of the scrollbar with the solid background color (that cannot be seen), changing the appearence of the scrollbarss.

Additionally, and more severe, it also seems to display the background color or some variation of it behind the signs besides the scrollbar, instead of just beeing transparent, such that they look weird.

This also happens if I use e.g. the Elflord-colorscheme via "colorscheme elflord", and remove the background via "highlight Normal guibg = none", and if I do not use Windows Terminal, but the "Windows Powershell" app of Win 10. I do not have a linux system lying around right now to also test this there.

![grafik](https://github.com/dstein64/nvim-scrollview/assets/18484746/a72f750a-75b2-47fe-b417-bd37eb7e8848)
![grafik](https://github.com/dstein64/nvim-scrollview/assets/18484746/83424f50-dcb8-4552-833a-bbb06c3fc4ef)

#
Elflord with "highlight Normal guibg = none":
![grafik](https://github.com/dstein64/nvim-scrollview/assets/18484746/ab451a3e-a427-4d67-8de1-fc05c27f71f6)

**Comment by dstein64** [comment]
Thanks for reporting the issue @HarthTuwist.

Do you have a minimal set of steps I can use to reproduce the problem? For example, something like:

1. Open Windows Terminal
2. Launch Neovim with `nvim` (with `nvim-scrollview` installed).
3. Make the terminal partially transparent by ...
4. Run `highlight Normal ...` to ...
5. ...

**Comment by HarthTuwist** [comment]
Hey, thanks for your fast answer! I replicated it wih the following init.vim. nvim-scrollview is installed via the standard plug way.

```
call plug#begin()

Plug 'dstein64/nvim-scrollview', { 'branch': 'main' }

call plug#end()


lua << EOF

local api = vim.api
local scrollview = require('scrollview')
scrollview.setup({
  winblend = 50,
  signs_on_startup = {'cursor', 'conflicts', 'marks', 'quickfix', 'search', 'diagnostics' },
  signs_column = 1,
  signs_overflow = 'right',
  column = 4,
  search_symbol = '*';
})

EOF

colorscheme elflord
highlight normal guibg = none
set termguicolors
```


At first, I had some trouble replicating, because I think the bug doesn't happen if `termguicolors` isn't set. The bug does not seem to apear on standard ctermg colors. If you use `highlight normal ctermbg=none` without, the bug does not appear for me. Hope that answers your questions.

![grafik](https://github.com/dstein64/nvim-scrollview/assets/18484746/ded60ce1-76ab-4e2b-bb7a-3825d1a66b9c)

**Comment by dstein64** [comment]
I reported the issue in https://github.com/neovim/neovim/issues/24584 (closed as a duplicate of https://github.com/neovim/neovim/issues/18576).

As a workaround for the signs, I've updated the code so that if there is a sign symbol, `winblend` will be turned off (non-whitespace characters can't be blended, so this shouldn't disable any functionality).
[7fa516c1bd73c0be3e4a8e6693a249fc6e85cb83](https://github.com/dstein64/nvim-scrollview/commit/7fa516c1bd73c0be3e4a8e6693a249fc6e85cb83)

Does that address the incorrect sign background colors you're seeing?

I think the coloring issue you reported for scrollbars is from the same underlying issue reported in https://github.com/neovim/neovim/issues/24584 and https://github.com/neovim/neovim/issues/18576. For that I don't have a workaround since setting `winblend=0` would effectively remove transparent scrollbar functionality.

**Comment by HarthTuwist** [comment]
Hey, sorry for responding so late! Indeed, updating my nvim-scrollview now fixes the bug with the signs!

Maybe it would be possible to set the scrollbar color such that the desired color results out of the blend of the background color? So if I want grey #808080, and the background color is black, I could set the scrollbar color to #FFFFFF, and it would result in the desired grey color? I haven't tested this, maybe blending works more complicated though.

Not sure if this will be worth the hazzle for you though, as there will be limits (for example, If the scrollbar should be white, and the background is black, there I think is no way to set the color to a value greater than white). Maybe just mention that as a known issue, and mention that people can change the color manually to take that into account?


Thanks for the nice support in general!

**Comment by dstein64** [comment]
Hi @HarthTuwist, for the winblend/highlighting interaction, I don't plan to have any scrollview-specific way of accommodating. For now, it's up to the user to carefully adjust the settings to control for how winblend and highlighting interact (the plugin is not doing anything extra here, but rather depending on Neovim's default behavior). I've added a comment to the documentation (640cc50532ca71627646798944a7a31dff2c6d78).

I don't know of a way to prevent the plugin from rendering over the buffer. As far as I know there is no built-in way to move the buffer left as you've described.

I can reproduce the issue you mentioned about the scrollbar not rendering properly under signs (when setting the sign column to 0). Thanks for letting me know. I've addressed this in ee97b36a13093faf442505d69da2e1a8ad8f5d97. For the mouse issue, I've updated the hovering behavior so that only the sign highlight will be toggled when hovering (not the bar), so that it doesn't suggest the scrollbar can be dragged. `:set mousemoveevent` is necessary to observe the hovering effect.
